sr_data_visualization/scripts/sr_data_visualization/sr_data_visualizer_gui.py

```diff
@@ -0,0 +1,1095 @@
+            self.radio_button_list.append(tmp)
+            j += 1
+
+        self.radio_button_position.toggled.connect(lambda: self.radio_button_list[0](self.radio_button_position))
+        self.radio_button_velocity.toggled.connect(lambda: self.radio_button_list[1](self.radio_button_velocity))
+        self.radio_button_effort.toggled.connect(lambda: self.radio_button_list[2](self.radio_button_effort))
+        self.radio_button_all.toggled.connect(lambda: self.radio_button_list[3](self.radio_button_all))
+
+        self.radio_button_setpoint.toggled.connect(lambda: self.radio_button_list[4](self.radio_button_setpoint))
+        self.radio_button_input.toggled.connect(lambda: self.radio_button_list[5](self.radio_button_input))
+        self.radio_button_dInput.toggled.connect(lambda: self.radio_button_list[6](self.radio_button_dInput))
+        self.radio_button_error.toggled.connect(lambda: self.radio_button_list[7](self.radio_button_error))
+        self.radio_button_output.toggled.connect(lambda: self.radio_button_list[8](self.radio_button_output))
+        self.radio_button_ctrl_all.toggled.connect(lambda: self.radio_button_list[9](self.radio_button_ctrl_all))
+
+        self.radioButton_strain_gauge_left.toggled.connect(
+            lambda: self.radio_button_list[10](self.radioButton_strain_gauge_left))
+        self.radioButton_strain_gauge_right.toggled.connect(
+            lambda: self.radio_button_list[11](self.radioButton_strain_gauge_right))
+        self.radioButton_measured_pwm.toggled.connect(
+            lambda: self.radio_button_list[12](self.radioButton_measured_pwm))
+        self.radioButton_measured_current.toggled.connect(
+            lambda: self.radio_button_list[13](self.radioButton_measured_current))
+        self.radioButton_measured_voltage.toggled.connect(
+            lambda: self.radio_button_list[14](self.radioButton_measured_voltage))
+        self.radioButton_measured_effort.toggled.connect(
+            lambda: self.radio_button_list[15](self.radioButton_measured_effort))
+        self.radioButton_temperature.toggled.connect(
+            lambda: self.radio_button_list[16](self.radioButton_temperature))
+        self.radioButton_unfiltered_position.toggled.connect(
+            lambda: self.radio_button_list[17](self.radioButton_unfiltered_position))
+        self.radioButton_unfiltered_force.toggled.connect(
+            lambda: self.radio_button_list[18](self.radioButton_unfiltered_force))
+        self.radioButton_last_commanded_effort.toggled.connect(
+            lambda: self.radio_button_list[19](self.radioButton_last_commanded_effort))
+        self.radioButton_encoder_position.toggled.connect(
+            lambda: self.radio_button_list[20](self.radioButton_encoder_position))
+        self.radioButton_all_motor_stat.toggled.connect(
+            lambda: self.radio_button_list[21](self.radioButton_all_motor_stat))
+
+    def make_all_button_functions(self, i, all, type):
+        if type == 0:
```

Comment:
I think you can implement this better as a dict

Reply:
done, much nicer that way, thanks :)